pandas/tests/frame/test_operators.py

```diff
@@ -804,13 +805,15 @@ def test_boolean_comparison(self):
         result = df.values > b
         assert_numpy_array_equal(result, expected.values)
 
-        result = df > l
-        assert_frame_equal(result, expected)
+        with pytest.raises(ValueError):
+            # wrong shape
+            df > l
 
-        result = df > tup
-        assert_frame_equal(result, expected)
+        with pytest.raises(ValueError):
+            # wrong shape
+            result = df > tup
 
-        result = df > b_r
+        result = df > b_r  # broadcasts like ndarray (GH#23000)
```

Comment:
Changed following #23000 (previously in this PR this comparison raised ValueError)

Reply:
can you put comments on the previous line